docs/en_US/Compressor/Pruner.md

```diff
@@ -177,6 +180,27 @@ The experiments code can be found at [examples/model_compress]( https://github.c
 
 ***
 
+## Constrained L1Filter Pruner
+This is a topology constraint-aware one-shot pruner. Compared to the [original L1 Filter Pruner](#l1filter-pruner), this pruner prunes the model not only based on the l1 norm of each filter, but also the topology of the network architecture of the target model. Specifically, for the example, if the output channels of two convolutional layers(conv1, conv2) are added together, then we can say that these two conv layers have channel dependency with each other(more details please see [Compression Utils](./CompressionUtils.md)). If we prune the first 50% of output channels(filters) for conv1, and prune the last 50% of output channels for conv2. Although both layers have pruned 50% of the filters, the speedup module still needs to add zeros to align the output channels. In this case, we cannot harvest the speed benefit from the model pruning. To better gain the speed benefit of the model pruning, we develop this constraint(topology)-aware one-shot pruner.
+
+The `Constrained L1Filter Pruner` will try to prune the same output channels for the layers that have the channel dependencies with each other. `Constrained L1Filter Pruner` will calculate the L1 norm sum of all the layers in the dependency set for each channel. We know that the maximum sparsity of the channels of this dependency set is determined by the minimum sparsity of layers in this dependency set(denoted by `min_sparsity`). According to the L1 norm sum of each channel, `Constrained L1Filter Pruner` will prune the same `min_sparsity` channels for all the layers. Next, the pruner will additionally prune `sparsity` - `min_sparsity` channels for each convolutional layer based on its own L1 norm of each channel. For example, suppose the output channels of `conv1` , `conv2` are added together and the configured sparsities of `conv1` and `conv2` are 0.3, 0.2 respectively. In this case, `Constrained L1Filter Pruner` will prune the same 20% of channels for `conv1` and `conv2` according to L1 norm sum of `conv1` and `conv2`. Next, the pruner will additionally prune 10% channels for `conv1` according to the L1 norm of each channel of `conv1`.
```

Comment:
Per this description, this feature is an enhancement to current pruners as my understanding.  So I suggest not to provide a new interface (`Constrained_L1FilterPruner`) to customer, better to use a switch/parameter in existing pruners `__init__` to turn on this feature.
Considering our current pruners prune filters/output channels for layer, there is another method for structured pruning is to prune the input channel at each layer which will result we need to check dependency reversely.
Maybe use following parameter ?
```
dependency_type:
    None:  current existing pruners
    filters:  provided by this PR
```
In the future we could support dependency_type to be `channels`

Reply:
Good suggestion! I'll update the interface. Thanks~

Reply:
Updated~ Thanks~